### ISSUE TITLE
Try and reduce deduplication in asm-parser-ewavr

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#4689]
### Is your feature request related to a problem? Please describe

It's a copy paste of vc parser with some (admittedly subtle) changes. Would be great to go over and see if there's anything _beneficial_ to deduplicate.

### Describe the solution you'd like

as above

### Describe alternatives you've considered

Leaving it alone, and removing the TODO

### Additional context

Not applicable

**Comment by partouf** [comment]
The problem with this is that ewavr and ewarm are proprietary and there’s no way to keep them supported. So it will need some kind of strategy.

**Comment by mattgodbolt** [comment]
fair enough; this is just a TODO as I just typescriptified vc6 and vc and ewarm. I was able to share most of the implementation between vc6 and vc, but I felt bad leaving it "nearly" the same.

**Comment by deepu-bansal** [comment]
hey , in which file I will get the code so as to solve the issue?

**Comment by nilaygit-10721** [comment]
can you assign me this issue?
